homeassistant/components/tradfri/cover.py

```diff
@@ -29,120 +24,51 @@
         async_add_entities(TradfriCover(cover, api, gateway_id) for cover in covers)
 
 
-class TradfriCover(CoverDevice):
+class TradfriCover(TradfriBaseDevice, CoverDevice):
     """The platform class required by Home Assistant."""
 
-    def __init__(self, cover, api, gateway_id):
+    def __init__(self, device, api, gateway_id):
         """Initialize a cover."""
-        self._api = api
-        self._unique_id = f"{gateway_id}-{cover.id}"
-        self._cover = None
-        self._cover_control = None
-        self._cover_data = None
-        self._name = None
-        self._available = True
-        self._gateway_id = gateway_id
+        super().__init__(device, api, gateway_id)
+        self._unique_id = f"{gateway_id}-{device.id}"
 
-        self._refresh(cover)
+        self._refresh(device)
 
     @property
     def supported_features(self):
         """Flag supported features."""
         return SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
 
-    @property
-    def unique_id(self):
-        """Return unique ID for cover."""
-        return self._unique_id
-
-    @property
-    def device_info(self):
-        """Return the device info."""
-        info = self._cover.device_info
-
-        return {
-            "identifiers": {(DOMAIN, self._cover.id)},
-            "name": self._name,
-            "manufacturer": info.manufacturer,
-            "model": info.model_number,
-            "sw_version": info.firmware_version,
-            "via_device": (DOMAIN, self._gateway_id),
-        }
-
-    async def async_added_to_hass(self):
-        """Start thread when added to hass."""
-        self._async_start_observe()
-
-    @property
-    def available(self):
-        """Return True if entity is available."""
-        return self._available
-
-    @property
-    def should_poll(self):
-        """No polling needed for tradfri cover."""
-        return False
-
-    @property
-    def name(self):
-        """Return the display name of this cover."""
-        return self._name
-
     @property
     def current_cover_position(self):
         """Return current position of cover.
 
         None is unknown, 0 is closed, 100 is fully open.
         """
-        return 100 - self._cover_data.current_cover_position
+        return 100 - self._device_data.current_cover_position
 
     async def async_set_cover_position(self, **kwargs):
         """Move the cover to a specific position."""
-        await self._api(self._cover_control.set_state(100 - kwargs[ATTR_POSITION]))
+        await self._api(self._device_control.set_state(100 - kwargs[ATTR_POSITION]))
 
     async def async_open_cover(self, **kwargs):
         """Open the cover."""
```

Comment:
@MartinHjelmare haven’t I implemented these methods wrong? It should be 100 for open?

Reply:
Home assistant thinks that 100 is open, yes.

https://github.com/home-assistant/home-assistant/blob/8bd847ed3969e75a1c03dcf652df28ab65106d1e/homeassistant/components/cover/__init__.py#L188-L194

Reply:
During the previous testing it was found the tradfri thinks that 0 is open and 100 is closed. So we invert.

Reply:
There have been some reports of the blinds not working properly, hence the question. But I've figured out the reason, I believe, and it's unrelated to this method.